modules/tensor_mechanics/src/kernels/StressDivergenceTensors.C

```diff
@@ -31,15 +29,22 @@ StressDivergenceTensors::StressDivergenceTensors(const std::string & name, Input
     _stress(getMaterialPropertyByName<RankTwoTensor>(_base_name + "stress")),
     _Jacobian_mult(getMaterialPropertyByName<ElasticityTensorR4>(_base_name + "Jacobian_mult")),
     _component(getParam<unsigned int>("component")),
-    _xdisp_coupled(isCoupled("disp_x")),
-    _ydisp_coupled(isCoupled("disp_y")),
-    _zdisp_coupled(isCoupled("disp_z")),
+    _ndisp(coupledComponents("displacements")),
+    _disp(3),
```

Comment:
The case when we have two displacements ```_ndisp = 2``` but still need to zero out the third displacement value (e.g. disp_z) breaks the code with a malloc error backtracing up into the Moose input parameters.  I asked John for help with the error, and he wasn't quite sure how to fix the problem.  I know that hardcoding to 3 is not very "moosey", and I'd appreciate help on solving the malloc error that occurs when using the code you suggested.